### PR TITLE
[KEYCLOAK-11345] - Test basic features of Keycloak.X with current testsuite

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

**Comment by pedroigor** [comment]
@dteleguin The initial work to support the Quarkus Server in our test suite. We've changed the context-root to `/auth`, reason being that the corresponding configuration property is a build time config and can not be changed when booting the server.

**Review by stianst — CHANGES_REQUESTED** [review]

File: distribution/server-x/src/main/content/bin/kc.sh [hunk 1]
```diff
@@ -20,6 +20,7 @@ if [ "x$RESOLVED_NAME" = "x" ]; then
     RESOLVED_NAME="$0"
 fi
 
+GREP="grep"
```
[the remark this answers is not in the document]
Comment (pedroigor) [reply distribution/server-x/src/main/content/bin/kc.sh]: Yes, it is. See line #62.

[the remark this answers is not in the document]
Comment (stianst) [reply distribution/server-x/src/main/content/bin/kc.sh]: I see. So it was missing ;)

File: services/src/main/java/org/keycloak/authentication/authenticators/challenge/NoCookieFlowRedirectAuthenticator.java [hunk 2]
```diff
@@ -45,7 +46,8 @@ public void authenticate(AuthenticationFlowContext context) {
 
         // only do redirects for GET requests
         if (HttpMethod.GET.equalsIgnoreCase(httpRequest.getHttpMethod())) {
-            if (!httpRequest.getUri().getQueryParameters().containsKey(LoginActionsService.AUTH_SESSION_ID)) {
+            KeycloakUriInfo uriInfo = context.getSession().getContext().getUri();
```
[the remark this answers is not in the document]
Comment (pedroigor) [reply services/src/main/java/org/keycloak/authentication/authenticators/challenge/NoCookieFlowRedirectAuthenticator.java]: I can send a separate PR, I thought that this change was too simple to deserve one.

Basically, we get some errors when calling `getUri` on the `httpRequest` when running Quarkus. I did not investigate further the reasons but I suspect some mismatch in resteasy versions.

File: distribution/server-x/pom.xml [hunk 3]
```diff
@@ -71,6 +71,7 @@
                                     <groupId>org.keycloak</groupId>
                                     <artifactId>keycloak-themes</artifactId>
                                     <outputDirectory>target/unpacked-themes</outputDirectory>
+                                    <excludes>**/node_modules/**,**/node/**</excludes>
```
[the remark this answers is not in the document]
Comment (pedroigor) [reply distribution/server-x/pom.xml]: I did not want to touch the main distribution so I thought to remove this from here. I think that centralize changes in Quarkus dist (even though they may not be the best) will help us in the future when reviewing what we have done.

[the remark this answers is not in the document]
Comment (stianst) [reply distribution/server-x/pom.xml]: Not sure why "node" is there in themes jar. That is a bug if it is.

There's node_modules that should be in the dist, and without it things won't actually work.

File: services/src/main/java/org/keycloak/authentication/authenticators/challenge/NoCookieFlowRedirectAuthenticator.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (stianst) [reply services/src/main/java/org/keycloak/authentication/authenticators/challenge/NoCookieFlowRedirectAuthenticator.java]: OK

File: distribution/server-x/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (stianst) [reply distribution/server-x/pom.xml]: Just tried building from master, and I don't have any "node" directories within the server-x distro, and I only have the expected node_modules directories:

    find -name node_modules
    ./keycloak-preview/account/resources/node_modules
    ./keycloak/common/resources/node_modules

[the remark this answers is not in the document]
Comment (pedroigor) [reply distribution/server-x/pom.xml]: Maybe something that was fixed. I'm going to remove.

File: quarkus/server/pom.xml [hunk 4]
```diff
@@ -402,6 +402,30 @@
...
             </exclusions>
         </dependency>
+        <dependency>
+            <groupId>com.sun.mail</groupId>
+            <artifactId>jakarta.mail</artifactId>
+            <version>${jakarta.mail.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.logging</groupId>
+            <artifactId>commons-logging-jboss-logging</artifactId>
```
Comment (dteleguin) [inline quarkus/server/pom.xml]: we have this one in the POM already (see line 204)

File: quarkus/server/pom.xml [hunk 5]
```diff
@@ -402,6 +402,30 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.resteasy</groupId>
+            <artifactId>resteasy-multipart-provider</artifactId>
```
Comment (dteleguin) [inline quarkus/server/pom.xml]: same here (line 200)

File: testsuite/integration-arquillian/servers/auth-server/quarkus/pom.xml [hunk 6]
```diff
@@ -0,0 +1,106 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>integration-arquillian-servers-auth-server</artifactId>
+        <groupId>org.keycloak.testsuite</groupId>
+        <version>9.0.0-SNAPSHOT</version>
```
Comment (dteleguin) [inline testsuite/integration-arquillian/servers/auth-server/quarkus/pom.xml]: need to bump version after rebase?

**Review by stianst — APPROVED** [review]

**Pull request merged**